src/connectors/refinement-list/connectRefinementList.ts

```diff
@@ -51,6 +51,10 @@ export type RefinementListItem = {
    * Indicates if the list item is refined.
    */
   isRefined: boolean;
+  /**
+   * List item childrens.
+   */
+  data?: RefinementListItem[];
```

Comment:
only hierarchical menu has `data`, not refinementList

Reply:
But hierarchical menu also uses the `RefinementList` component so the `RefinementListItem` received can either have a data property or not.

What I could do is to split this type in 2 (eg. `RefinementListItem` and `HierarchicalRefinementListItem`) and update the `RefinementList` component to accept the 2 types, so the `data` property does not leak into `connectRefinementList`. I'll do this in the HierachicalMenu PR since it needs to be coverted to TS first.

Reply:
I think there should be a type in the refinement list component that indeed declares the item with data optional, and a separate item type in both connect refinement list & connect hierarchical menu

Reply:
Done in https://github.com/algolia/instantsearch.js/pull/4711 (split it in 2 types `RefinementListItem` and `HierarchicalMenuItem`)